### PR TITLE
Using empty() to check for a set db

### PR DESCRIPTION
**Pull request by ryanml**
Signed-off-by: Ryan Lanese <rlanese@asu.edu>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

### CODE REVIEW

**Comment by ryanml** [comment]
@mauriciofauth I opened a new PR using empty, as I made I pushed a few incorrect commits to the old one by mistake (I figured this was easier) [ref PR #13698]

I typically haven't used empty due to differences in behavior across PHP versions, but you're right in that it's better for readability, and will likely be fine for the purposes of this fix.

**Comment by ryanml** [comment]
Also to note, this will fix issue #12856 as well as #13490

**Comment by mauriciofauth** [comment]
This change broke one of the [Selenium tests](https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/278190572). See:
```
1) PMA_SeleniumImportTest::testServerImport
PHPUnit_Extensions_Selenium2TestCase_WebDriverException: no such element: Unable to locate element: {"method":"xpath","selector":"//div[@class='success' and contains(., 'Import has been successfully')]"}
```

**Comment by ryanml** [comment]
@mauriciofauth I imagine the test case would have to change a bit to accommodate this behavior

I think a check like this would have to happen in PmaSeleniumImportTest.php on line 112:

`if ($this->isElementPresent("byXPath", "//a[@class='item' and contains('Database: ')]"))`

From the dom perspective, this looks at the top nav for the presence of a selected database, and ideally the test would only expect the success message if that exists

Does this seem like a reasonable adjustment to you? I appreciate your patience :)

**Comment by mauriciofauth** [comment]
What do you think about this, @devenbansod?

**Comment by mauriciofauth** [comment]
@ryanml Feel free to make changes to the tests. :)

**Comment by ryanml** [comment]
@mauriciofauth will do :) I'll be able to get to it this evening, thanks!

**Comment by mauriciofauth** [comment]
@ryanml about the new PR. If you push some incorrect commits or just minor fixes to previous commits, you can squash them with `git rebase`.

**Comment by ryanml** [comment]
@mauriciofauth thanks for the tips :)

It looks like everything is checking out with the selenium tests through Travis, are those code coverage checks anything I need to be concerned about?

**Comment by devenbansod** [comment]
[bs-video-logs-use.mp4.zip](https://github.com/phpmyadmin/phpmyadmin/files/1323468/bs-video-logs-use.mp4.zip)
I could not find time to go through the code completely, but this condition of database being empty might actually be true for a server import.

Also, as you can see the attached video, it actually gives an error in the test. So adding the if condition in the test does not seem like a good idea.

**Comment by ryanml** [comment]
@devenbansod the adjustment I made in the test will work fine when errors are shown, the issue is that the coverage didn't account for my code change, that a success message should only be show when a database is selected

On that note, if it is true for a server import, I'll be happy to withdraw this PR. However, any time I attempt any import without a db selected, MySQL throws a 1046, that and previous issues leads me to believe it's a constant. So just let me know with this one, thanks :)

cc: @mauriciofauth

File: import.php [hunk 1]
```diff
@@ -573,7 +573,7 @@
 } elseif ($finished && ! $error) {
     // Do not display the query with message, we do it separately
     $display_query = ';';
-    if ($import_type != 'query') {
+    if ($import_type != 'query' && !empty($db)) {
```
Comment (nijel) [inline import.php]: Using empty() is usually not a good idea, imagine database called 0 (zero).

Comment (ryanml) [reply import.php]: @nijel Originally I had done ...&& $db != '', after @mauriciofauth suggested using empty().

Would the former be better, or may I have to do some more stringent checking to be sure that we in fact have an empty database? Eg maybe something like strlen($db) > 0, or perhaps use isset() and another validator

I'm glad to continue working on this, thank you both mauricio and nigel for your help

Comment (mauriciofauth) [reply import.php]: I don't know why, but I was sure that 0 was an invalid name. Anyway, we have to check the problem pointed out by Deven.

Comment (ryanml) [reply import.php]: I would have thought the same thing too, and cool. I'll check it out more as well

Comment (nijel) [reply import.php]: Actually 0 is valid database and table name (and quite lot of things in phpMyAdmin break with it exactly due to using empty as check in many places).

**Comment by ryanml** [comment]
@nijel @mauriciofauth I have pushed another commit again making the check ($db != '')

In any situation a database is not chosen, $db always contains an empty string.

This probably the best solution for what this PR is trying to accomplish, let me know what you guys think and if you'd like to move forward with this :)

**Comment by ryanml** [comment]
@nijel have you taken a look at this since my last commit or have any final thoughts on the issue in general?

**Comment by nijel** [comment]
Honestly I'm not really sure what are you trying to achieve here. Why the server import should behave differently here?

**Comment by mauriciofauth** [comment]
It's related to #12856 and #13490.

**Comment by nijel** [comment]
I think we should not show the success message on error, regardless it is on server, database or table. Not showing success message at all sounds confusing.

**Comment by ryanml** [comment]
@nijel Sorry, I'm a little confused by your response, do you think we should, or should not show the success message? The standing issue was that the success message implies that a _successful_ action has taken place

**Comment by nijel** [comment]
We should not show the success message if there was error during the import. The fact whether $db is set or not is really not relevant to that.

**Comment by nijel** [comment]
To make it more explicit: The error handling is broken in all import situations, not only server wide.

screenshot-2017-10-3 localhost localhost gammu phpmyadmin 4 8 0-dev

**Comment by nijel** [comment]
Sorry, but this is wrong approach to the issue (which was wrongly described as well). See https://github.com/phpmyadmin/phpmyadmin/issues/12856#issuecomment-333875265

**Pull request closed without merging**